vxquery-core/src/main/java/org/apache/vxquery/runtime/functions/util/FunctionHelper.java

```diff
@@ -487,316 +487,320 @@ public static boolean arraysEqual(byte[] bytes1, int offset1, int length1, byte[
-                            tvp2.getValue(tp2.datep);
-                            return aOp.operateGYearGYear(tp1.datep, tp2.datep, dCtx);
-                    }
-                    break;
-
-                case ValueTag.XS_HEX_BINARY_TAG:
-                    tvp1.getValue(tp1.binaryp);
-                    switch (tid2) {
-                        case ValueTag.XS_HEX_BINARY_TAG:
-                            tvp2.getValue(tp2.binaryp);
-                            return aOp.operateHexBinaryHexBinary(tp1.binaryp, tp2.binaryp);
-                    }
-                    break;
-
-                case ValueTag.XS_BASE64_BINARY_TAG:
-                    tvp1.getValue(tp1.binaryp);
-                    switch (tid2) {
-                        case ValueTag.XS_BASE64_BINARY_TAG:
-                            tvp2.getValue(tp2.binaryp);
-                            return aOp.operateBase64BinaryBase64Binary(tp1.binaryp, tp2.binaryp);
-                    }
-                    break;
-
-                case ValueTag.XS_ANY_URI_TAG:
-                    tvp1.getValue(tp1.utf8sp);
-                    switch (tid2) {
-                        case ValueTag.XS_ANY_URI_TAG:
-                            tvp2.getValue(tp2.utf8sp);
-                            return aOp.operateAnyURIAnyURI(tp1.utf8sp, tp2.utf8sp);
-
-                        case ValueTag.XS_STRING_TAG:
-                        case ValueTag.XS_NORMALIZED_STRING_TAG:
-                        case ValueTag.XS_TOKEN_TAG:
-                        case ValueTag.XS_LANGUAGE_TAG:
-                        case ValueTag.XS_NMTOKEN_TAG:
-                        case ValueTag.XS_NAME_TAG:
-                        case ValueTag.XS_NCNAME_TAG:
-                        case ValueTag.XS_ID_TAG:
-                        case ValueTag.XS_IDREF_TAG:
-                        case ValueTag.XS_ENTITY_TAG:
-                        case ValueTag.XS_UNTYPED_ATOMIC_TAG:
-                            tvp2.getValue(tp2.utf8sp);
-                            return aOp.operateStringString(tp1.utf8sp, tp2.utf8sp);
-                    }
-                    break;
-
-                case ValueTag.XS_QNAME_TAG:
-                    tvp1.getValue(tp1.qnamep);
-                    switch (tid2) {
-                        case ValueTag.XS_QNAME_TAG:
-                            tvp2.getValue(tp2.qnamep);
-                            return aOp.operateQNameQName(tp1.qnamep, tp2.qnamep);
-                    }
-                    break;
-
-                case ValueTag.XS_NOTATION_TAG:
-                    tvp1.getValue(tp1.utf8sp);
-                    switch (tid2) {
-                        case ValueTag.XS_NOTATION_TAG:
-                            tvp2.getValue(tp2.utf8sp);
-                            return aOp.operateNotationNotation(tp1.utf8sp, tp2.utf8sp);
-                    }
-                    break;
+            if (tid1 == ValueTag.JS_NULL_TAG || tid2 == ValueTag.JS_NULL_TAG) {
```

Comment:
Please add this tag to the switch statement. The first switch is for the first argument and the second switch is for the second argument.